### PR TITLE
[GH-#267] check and fix dialyzer issues, include in ci

### LINKED ISSUE #267: check and fix dyalizer issues, include in ci

### PR DESCRIPTION
**Pull request by 0legIv**
resolves #267

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (cytadela8) [inline .vscode/launch.json]: Should this file be here?

File: apps/aehttpclient/lib/client.ex [hunk 2]
```diff
@@ -16,7 +16,7 @@ defmodule Aehttpclient.Client do
   @typedoc "Client request identifier"
   @type req_kind :: :default | :pool_txs | :acc_txs | :info | :block | :raw_blocks
 
-  @spec handle_response(req_kind(), map(), list(map())) :: {:ok, map()}
+  @spec handle_response(req_kind(), any(), list(map())) :: {:ok, any()}
```
Comment (thepiwo) [inline apps/aehttpclient/lib/client.ex]: can you alias any to make the type definition semantic correct

File: .vscode/launch.json (hunk 1 above)
Comment (thepiwo) [reply .vscode/launch.json]: @0legIv you can add this to the gitignore

File: apps/aecore/lib/aecore/chain/worker.ex [hunk 3]
```diff
@@ -195,8 +195,8 @@ defmodule Aecore.Chain.Worker do
     Enum.reverse(get_blocks([], start_block_hash, final_block_hash, count))
   end
 
-  @spec get_block_by_height(non_neg_integer(), binary() | nil) ::
-          Chainstate.account_chainstate() | {:error, binary()}
+  @spec get_chain_state_by_height(non_neg_integer(), binary() | nil) ::
+          ChainState.account_chainstate() | {:error, binary()}
```
Comment (gspasov) [inline apps/aecore/lib/aecore/chain/worker.ex]: The return value should be `Chainstate.t()`

File: apps/aehttpclient/lib/client.ex [hunk 4]
```diff
@@ -16,7 +16,7 @@ defmodule Aehttpclient.Client do
   @typedoc "Client request identifier"
   @type req_kind :: :default | :pool_txs | :acc_txs | :info | :block | :raw_blocks
 
-  @spec handle_response(req_kind(), map(), list(map())) :: {:ok, map()}
+  @spec handle_response(req_kind(), map() | list(), list(map())) :: {:ok, map()}
```
Comment (gspasov) [inline apps/aehttpclient/lib/client.ex]: Put the specification above it's functions

File: apps/aeutil/lib/serialization.ex [hunk 5]
```diff
@@ -212,8 +212,8 @@ defmodule Aeutil.Serialization do
   @doc """
   Initializing function to the recursive functionality of deserializing a strucure
   """
-  @spec deserialize_value(any()) :: any()
-  def deserialize_value(value), do: deserialize_value(value, "")
+  @spec deserialize_value(any(), atom()) :: any()
```
Comment (gspasov) [inline apps/aeutil/lib/serialization.ex]: Fix the specifications of the deserialize_value() and move each specification to it's own function.

**Comment by thepiwo** [comment]
@0legIv will you still respond to my prior comment? https://github.com/aeternity/elixir-node/pull/354#pullrequestreview-113868494

also is `:dialyzer.run error: No such file, directory or application: "_build/dev/lib/aecore/ebin"` a problem (see travis)?

**Comment by 0legIv** [comment]
@thepiwo 
Is it okay to squash and merge the whole branch?

**Comment by thepiwo** [comment]
yes

**Comment by 0legIv** [comment]
@cytadela8 can you rereview, please?

**Comment by thepiwo** [comment]
@gspasov can you approve?

**Pull request merged**